Spigot-Server-Patches/0739-More-Lidded-Block-API.patch

```diff
@@ -0,0 +1,120 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: LemonCaramel <admin@caramel.moe>
+Date: Sun, 23 May 2021 17:49:51 +0900
+Subject: [PATCH] More Lidded Block API
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/TileEntityEnderChest.java b/src/main/java/net/minecraft/world/level/block/entity/TileEntityEnderChest.java
+index 2bc4213c70be47ca8bbc24898cc92e43f4228821..ff7ba8f98010d19048d96f6e3b5f7259f4728a17 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/TileEntityEnderChest.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/TileEntityEnderChest.java
+@@ -12,6 +12,7 @@ public class TileEntityEnderChest extends TileEntity { // Paper - Remove ITickab
+     public float b;
+     public int c;
+     private int g;
++    public boolean opened; // Paper - More Lidded Block API
+ 
+     public TileEntityEnderChest() {
+         super(TileEntityTypes.ENDER_CHEST);
+@@ -40,6 +41,13 @@ public class TileEntityEnderChest extends TileEntity { // Paper - Remove ITickab
+         double d0;
+         // Paper end
+ 
++        // Paper start - More Lidded Block API
++        if (opened) {
++            this.c--;
```

Comment:
Why decrement the viewer count here?

Reply:
Before calling the "doOpenLogic()" method, c is increased.

Reply:
If a player opens it, it should still increment the viewer count, otherwise you'd have desync issues between plugin calls to open/close here. You should instead return before both the block action and close/open logic are called in `d()` and `f()`

Reply:
oh, I referred and wrote this code for TileEntityChest. I checked that code again, and realized that i didn't need this code.
Thank you for letting me know.